schemas/theme/preset.json

```diff
@@ -0,0 +1,55 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Shopify Liquid Section or Block Preset Schema",
+  "oneOf": [
+    {
+      "$ref": "#/definitions/presetWithBlocksArray"
+    },
+    {
+      "$ref": "#/definitions/presetWithBlocksHash"
+    }
```

Comment:
I encountered an issue where presets with only `name` and `settings` could be validated by either of these refs, so by adding `blocks` as `required` to the `presetWithBlocksHash` it ensured there was no such conflict. A preset with only `name` (and usually `settings`, otherwise the whole preset's kinda useless) will get validated by the `presetWithBlocksArray`.

Optionally, we could make `blocks` required in `presetWithBlocksArray` too and require a third option for presets with just `name` and `settings`. A trivial change, if it's preferred. I'm on the fence, myself.

Reply:
maybe...
```json
"allOf": [
  { "$ref": "#/definitions/presetBase" }
  { "oneOf": [
    { "$ref": "#/definitions/presetBlockArray" },
    { "$ref": "#/definitions/presetBlockHash" },
  ]}
]
```
?

Reply:
So I remember when we chatted in person, we agreed it wasn't worth complicating things just to extract out `name` and `settings` - though I'd be down to do it if it solved the "only" `name` and `settings` more elegantly... Although the one you posted still requires that the data fit into either `presetBlocksArray` and `presetBlocksHash`? 

Wouldn't it be more like this to accommodate all 3 possible cases?
```
  { "oneOf": [
    { "$ref": "#/definitions/presetBase" }
    { "$ref": "#/definitions/presetBlocksArray" },
    { "$ref": "#/definitions/presetBlocksHash" },
  ]}
```

Reply:
I mean that works too so long as presetBase doesn't accept other properties and that the array/hash have blocks & blocks order required. Same same

Reply:
Cool. The change you proposed ended up being easier to implement - my version was giving me a hard time for a reason I can't figure out, so, if it's same same, then yours works :) 

Did some tophatting locally:
<img width="612" alt="image" src="https://github.com/user-attachments/assets/301d4954-4274-4501-804e-dcf926aec8d8">

(worked with blocks as arrays and blocks as hash)